WIP : Any number of channels

BadrYoubiIdrissi opened the pull request:
![build](https://travis-ci.com/BadrYoubiIdrissi/lucid.svg?branch=any_number_of_channels) [![Coverage Status](https://coveralls.io/repos/github/BadrYoubiIdrissi/lucid/badge.svg?branch=any_number_of_channels)](https://coveralls.io/github/BadrYoubiIdrissi/lucid?branch=any_number_of_channels)

Modified the image parametrization to accept any number of channels. 

The following choices were made :
- No decorrelation is done across channels. 
- A sigmoid is applied on all channels. 

On top of the main channels modification, there is a commit to fix a bug switching width and height.

colah commented on lucid/optvis/param/color.py:
```diff
@@ -73,3 +73,16 @@ def to_valid_rgb(t, decorrelate=False, sigmoid=True):
   else:
     return constrain_L_inf(2*t-1)/2 + 0.5
+
+def arbitrary_channels_to_rgb(*args, channels=None, **kwargs):
```
I'm not sure I understand when I'd use this function in practice. Is it just for the tests? Maybe add a comment. :)

BadrYoubiIdrissi replied:
Yes exactly :) It's a funky parametrization to test more than 3 channels

colah replied:
Makes sense! Consider adding a brief comment, or even moving into the tests? Otherwise, LGTM!

BadrYoubiIdrissi commented:
@colah It's done :) I have moved the `arbitrary_channels_to_rgb` function to the tests as you suggested alongside with an integration level test. I also added the `grayscale_image_rgb` to the param package init for direct access. Thank you for reviewing !
